perun-web-gui/src/main/java/cz/metacentrum/perun/webgui/tabs/entitylessattributestabs/EntitylessAttributeEditKeyTabItem.java

```diff
@@ -0,0 +1,331 @@
+
+		// MAIN PANEL
+		VerticalPanel firstTabPanel = new VerticalPanel();
+		firstTabPanel.setSize("100%", "100%");
+
+		// HORIZONTAL MENU
+		TabMenu menu = new TabMenu();
+		// refresh
+		menu.addWidget(UiElements.getRefreshButton(this));
+
+		// Get Attributes
+		final GetEntitylessAttributes jsonCallback = new GetEntitylessAttributes(def);
+
+		// get the table
+		CellTable<Attribute> table = jsonCallback.getTable();
+
+		this.map = jsonCallback.getMap();
+
+		if (!isAuthorized()) jsonCallback.setCheckable(false);
+
+		final CustomButton setButton = TabMenu.getPredefinedButton(ButtonType.SAVE, ButtonTranslation.INSTANCE.saveChangesInAttributes());
+		menu.addWidget(setButton);
+		if (!isAuthorized()) setButton.setEnabled(false);
+
+		// refresh table
+		final JsonCallbackEvents events = JsonCallbackEvents.refreshTableEvents(jsonCallback);
+
+		// set button event with button disable
+		final JsonCallbackEvents setButtonEvent = JsonCallbackEvents.disableButtonEvents(setButton, events);
+
+		setButton.addClickHandler(event -> {
+
+			ArrayList<Attribute> list = jsonCallback.getTableSelectedList();
+
+			if (UiElements.cantSaveEmptyListDialogBox(list)) {
+				Map<String, String> ids = new HashMap<>();
+				SetEntitylessAttribute request = new SetEntitylessAttribute(setButtonEvent);
+				for (Attribute a : list) {
+					ids.put("key", map.get(a));
+					request.setAttribute(ids, a);
+				}
+				draw();
+			}
+		});
+
+		final TextBox keyBox = new TextBox();
+		keyBox.getElement().setPropertyString("placeholder", "enter key for new attribute");
+
+		final TextBox valueBox = new TextBox();
+		valueBox.getElement().setPropertyString("placeholder", "enter value for new attribute");
+
+		//add attr button
+		CustomButton addButton = TabMenu.getPredefinedButton(ButtonType.ADD, true,
+			ButtonTranslation.INSTANCE.setNewAttributes(), event -> {
+				map = jsonCallback.getMap();
+				if (map.containsValue(keyBox.getText())) {
+					UiElements.generateAlert("", "Key \"" + keyBox.getText() + "\" already exists");
+				} else if (keyBox.getText().equals("")) {
+					UiElements.generateAlert("", "Enter key into key box");
+				} else if (valueBox.isVisible() && !valueBox.getText().equals("")) {
+					addAttribute(keyBox.getText(), valueBox.getText());
+					draw();
+				} else if (!valueBox.isVisible()) {
+					addAttribute(keyBox.getText(), valueBox.getText());
+					draw();
+				} else {
+					UiElements.generateAlert("", "Enter value into value box");
+				}
+			});
+		menu.addWidget(addButton);
+		if (!isAuthorized()) addButton.setEnabled(false);
+
+		// remove attr button
+		final CustomButton removeButton = TabMenu.getPredefinedButton(ButtonType.REMOVE, ButtonTranslation.INSTANCE.removeAttributes());
+		menu.addWidget(removeButton);
+		if (!isAuthorized()) removeButton.setEnabled(false);
+
+
+		// remove button event
+		final JsonCallbackEvents removeButtonEvent = JsonCallbackEvents.disableButtonEvents(removeButton, events);
```

Comment:
What is this event for?

Reply:
it was remnant of me wanting to disable remove button when no attribute is selected. I removed it